### PR TITLE
revert: fix(core): add validation for all the connector auth_type

### PR DESCRIPTION
**Pull request by Aprabhat19**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

This PR reverts the commit of `1cda7ad` (PR #1748 ) and also adds a new implementation for the validation of auth_type

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

The previous changes caused conflicting implementation with the connector flow of `signifyd`

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/authorizedotnet.rs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #![allow(dead_code)]
```
Comment (SanchithHegde) [inline crates/router/src/connector/authorizedotnet.rs:1]: Unrelated, can we remove this annotation?

File: crates/router/src/connector/checkout.rs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #![allow(dead_code)]
```
Comment (SanchithHegde) [inline crates/router/src/connector/checkout.rs:1]: Same here.

File: crates/router/src/core/admin.rs [hunk 3]
```diff
@@ -874,3 +871,210 @@ pub fn get_frm_config_as_secret(
...
+
+    match connector_name {
+        api_enums::Connector::DummyConnector1 => {
+            dummyconnector::transformers::DummyConnectorAuthType::try_from(val)?;
+            Ok(())
+        }
+        api_enums::Connector::DummyConnector2 => {
+            dummyconnector::transformers::DummyConnectorAuthType::try_from(val)?;
+            Ok(())
+        }
```
Comment (SanchithHegde) [inline crates/router/src/core/admin.rs:889]: You can merge all of the dummy connector arms.

File: crates/router/src/core/admin.rs [hunk 4]
```diff
@@ -874,3 +871,210 @@ pub fn get_frm_config_as_secret(
...
+        api_enums::Connector::Adyen => {
+            adyen::transformers::AdyenAuthType::try_from(val)?;
+            Ok(())
+        }
+        api_enums::Connector::Airwallex => {
+            airwallex::transformers::AirwallexAuthType::try_from(val)?;
+            Ok(())
+        }
+        api_enums::Connector::Authorizedotnet => {
+            authorizedotnet::transformers::MerchantAuthentication::try_from(val)?;
```
Comment (SanchithHegde) [inline crates/router/src/core/admin.rs:923]: @Narayanbhat166 Should we consider naming the type as `AuthorizedotnetAuthType` instead?

File: crates/router/src/core/admin.rs [hunk 5]
```diff
@@ -874,3 +871,210 @@ pub fn get_frm_config_as_secret(
...
+        api_enums::Connector::Tsys => {
+            tsys::transformers::TsysAuthType::try_from(val)?;
+            Ok(())
+        }
+        api_enums::Connector::Wise => {
+            wise::transformers::WiseAuthType::try_from(val)?;
+            Ok(())
+        }
+        api_enums::Connector::Worldline => {
+            worldline::transformers::AuthType::try_from(val)?;
```
Comment (SanchithHegde) [inline crates/router/src/core/admin.rs:1064]: @ArjunKarthik should we consider renaming this as `WorldlineAuthType`?

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

File: crates/router/src/core/admin.rs (hunk 5 above)
Comment (ArjunKarthik) [reply crates/router/src/core/admin.rs:1009]: Yeah @SanchithHegde

**Review by ArjunKarthik — APPROVED** [review]

**Review by Narayanbhat166 — DISMISSED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

File: crates/router/src/connector/adyen/transformers.rs [hunk 6]
```diff
@@ -54,14 +54,14 @@ pub enum AdyenRecurringModel {
 }
 
 #[derive(Clone, Default, Debug, Serialize, Deserialize)]
-pub enum AuthType {
+pub enum WorldlineAuthType {
```
Comment (ArjunKarthik) [inline crates/router/src/connector/adyen/transformers.rs:57]: It should be AdyenAuthType

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/adyen/transformers.rs (hunk 6 above)
Comment (ArjunKarthik) [reply crates/router/src/connector/adyen/transformers.rs:57]: This is not related to Connector Auth type, Just keep it as it is

**Review by ArjunKarthik — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Pull request merged**